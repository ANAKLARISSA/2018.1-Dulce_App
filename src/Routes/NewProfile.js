import React from 'react';
import NewProfileNavigator from '../Navigators/NewProfileNavigator';

class NewProfile extends React.Component {

  render() {
    return (
      <NewProfileNavigator />
    );
  }
}

export default NewProfile;
