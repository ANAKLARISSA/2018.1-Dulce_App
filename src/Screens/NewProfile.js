import React from 'react';
import {View,Text, Picker} from 'react-native';
import ScreenHeader from '../Components/ScreenHeader';
import t from 'tcomb-form-native';
import {Container} from 'native-base';
import SideBar from '../Components/SideBar';
import SignUpButton from '../Components/SignUpButton';
import {newProfile as styles} from './styles' ;

const default_profile_state_value = {
  value: {
    nome: '',
    setor: '',
    matricula: '',
    hospital: ''
  },
  userType: ''
};
const Form = t.form.Form;
const default_profile_options = {
  fields: {
    Nome: {
      error: 'Campo obrigatório'
    },
    Email: {
      error: 'Campo obrigatório'
    },
    senha: {
      secureTextEntry: true,
      error: 'Campo obrigatório'
    },
    confirmarSenha: {
      secureTextEntry: true,
      error: 'Campo obrigatório'
    }
  },
  stylesheet: styles.text
};
class NewProfile extends React.Component {
  constructor(props) {
    super(props);
    this.Service = t.struct({
      nome: t.String,
      setor: t.String,
      matricula: t.String,
      hospital: t.String
    });
    this.state = default_profile_state_value;
    this.options = default_profile_options;
  }

  onValueChange(value) {
    this.setState({
      selected: value
    });
  }

  pickerButton() {
    return (
      <Picker
        selectedValue={this.state.userType}
        style={{height: 50, width: 200,marginBottom: 5,marginLeft: 10}}
        onValueChange={(itemValue) => this.setState({userType: itemValue})}>
        <Picker.Item label='Funcionário' value='employee' />
        <Picker.Item label='Gestor do setor' value='sector_manager' />
        <Picker.Item label='gestor da instituição' value='institution_manager' />
      </Picker>
    );
  }

  onChange(value) {
    this.setState({selected: value});
  }

  render() {
    const {goBack} = this.props.navigation;
    return (
      <View style={{flexDirection: 'row', flex: 1}}>
        <SideBar />
        <Container style={{backgroundColor: '#FFF', flex: 8}}>
          <ScreenHeader title = 'Criar novo perfil' goBack = {() => goBack()} />
          <View style={styles.container}>
            <Form
              ref='form'
              type={this.Service}
              value={this.state.value}
              options={this.options}
              onChange={(v) => this.onChange(v)}
            />
          </View>
          <Text style={styles.text}> Selecione o tipo de Funcionário que deseja criar </Text>
          {this.pickerButton()}
          <View style={{flex: 1}}>
            <SignUpButton
              text = 'Cadastrar'
              onPress = {() => {}}/>
          </View>
        </Container>
      </View>
    );
  }
}

export default NewProfile;
