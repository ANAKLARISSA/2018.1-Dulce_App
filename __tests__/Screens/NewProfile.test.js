import React from 'react';
import renderer from 'react-test-renderer';
import NewProfile from '../../src/Screens/NewProfile';
import 'react-native';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import {shallow} from 'enzyme';
// import t from 'tcomb-form-native';

require('bezier');
Enzyme.configure({adapter: new Adapter()});

it('renders correctly', () => {
  const tree = renderer
  .create(<NewProfile navigation={{}} />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});

it('should test onChange correctly', () => {
// This is to test component functions
  let newProfile = renderer
  .create(
    <NewProfile
      navigation={{}}
      dispatch={action =>
        expect(action)
        .toEqual(true)
      }
    />
  )
  .getInstance();

  newProfile.onChange();
});

it('should test onValueChange correctly', () => {
// This is to test component functions
  let newProfile = renderer
  .create(
    <NewProfile
      navigation={{}}
      dispatch={action =>
        expect(action)
        .toEqual(true)
      }
    />
  )
  .getInstance();

  newProfile.onValueChange();
});

it('should test onValueChange correctly', () => {
// This is to test component functions
  let newProfile = renderer
  .create(
    <NewProfile
      navigation={{}}
      dispatch={action =>
        expect(action)
        .toEqual(true)
      }
    />
  )
  .getInstance();

  newProfile.onValueChange();
});
it('should test onValueChange correctly', () => {
// This is to test component functions
  let newProfile = renderer
  .create(
    <NewProfile
      navigation={{}}
      dispatch={action =>
        expect(action)
        .toEqual(true)
      }
    />
  )
  .getInstance();

  newProfile.onValueChange();
});

const flushPromises = () => new Promise(resolve => setImmediate(resolve));

it('Should call createProfile function', async() => {
  const navigation = jest.fn();
  const value = {
    matricula: '12345',
    especialidade: '12345'
  };
  const spy = jest.spyOn(NewProfile.prototype, 'createProfile');
  const wrapper = shallow(<NewProfile navigation = {navigation} />);
  console.log(wrapper.debug);
  wrapper.setState({value: value});
  await flushPromises();
  wrapper.update();
  const enterButton = wrapper.find('AGRButton').at(0);
  enterButton.simulate('press');
  expect(spy).toHaveBeenCalled();
});

// it('Should call goBack function newProfile', async() => {
//   const navigation = jest.fn();
//   const _goBack = jest.fn();
//   const spy = _goBack;
//   const wrapper = shallow(<NewProfile navigation = {navigation} />);
//   console.log(wrapper.debug());
//
//   const header = wrapper.find('ScreenHeader').at(0).dive();
//   const goback = header.find('Styled(Button)').at(0);
//   console.log(header.debug());
//   console.log(goback.simulate('press'));
//   // expect(spy).toHaveBeenCalled();
// });

// it('Should call selectHospital function', async() => {
//   const navigation = jest.fn();
//   const value = {
//     matricula: '12345',
//     especialidade: '12345'
//   };
//   // const hospitals = [
//   //   {
//   //     name: 'firstHospital',
//   //     id: '12345'
//   //   },
//   //   {
//   //     name: 'secondHospital',
//   //     id: '123456'
//   //   }
//   // ];
//   const spy = jest.spyOn(NewProfile.prototype, 'selectHospital');
//   const wrapper = shallow(<NewProfile navigation = {navigation} />);
//   // wrapper.setState({hospitals: hospitals, loadingHospital: false});
//   console.log(wrapper.debug());
//   wrapper.setState({value: value});
//   await flushPromises();
//   wrapper.update();
//   const selectHospital = wrapper.find('PickerItem').at(5);
//   selectHospital.simulate('press');
//   expect(spy).toHaveBeenCalled();
// });
