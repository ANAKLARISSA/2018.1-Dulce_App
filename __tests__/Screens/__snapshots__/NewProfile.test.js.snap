// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fd7d01",
        "flex": 0.25,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#f1f1f1",
          "borderRadius": 5,
          "height": 40,
          "margin": 1,
          "width": 7.5,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "height": 1334,
        },
        Object {
          "backgroundColor": "#FFF",
          "flex": 8,
        },
      ]
    }
    virtual={undefined}
  >
    <View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#F8F8F8",
              "borderBottomColor": "#a7a6ab",
              "borderBottomWidth": 0.5,
              "elevation": 3,
              "flexDirection": "row",
              "height": 64,
              "justifyContent": "center",
              "left": 0,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 15,
              "right": 0,
              "shadowColor": undefined,
              "shadowOffset": undefined,
              "shadowOpacity": undefined,
              "shadowRadius": undefined,
              "top": 0,
            },
            Object {
              "backgroundColor": "#5f4b8b",
            },
          ]
        }
        virtual={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "alignSelf": "center",
              "flex": 1,
            }
          }
          virtual={undefined}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": null,
                "backgroundColor": "transparent",
                "borderBottomWidth": null,
                "borderColor": null,
                "borderLeftWidth": null,
                "borderRadius": 5,
                "borderRightWidth": null,
                "borderTopWidth": null,
                "elevation": 0,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "space-between",
                "marginLeft": -3,
                "opacity": 1,
                "paddingBottom": 6,
                "paddingLeft": 6,
                "paddingRight": 6,
                "paddingTop": 6,
                "shadowColor": null,
                "shadowOffset": null,
                "shadowOpacity": null,
                "shadowRadius": null,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                    Object {
                      "color": "#007aff",
                      "fontSize": 30,
                      "marginLeft": 2,
                      "marginRight": 2,
                      "marginTop": 0,
                      "paddingTop": 2,
                    },
                  ],
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
              virtual={undefined}
            >
              Ôèè
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "flex": 1,
              },
              Object {},
            ]
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 17,
                  "fontWeight": "600",
                  "textAlign": "center",
                },
                Object {
                  "fontFamily": "Raleway",
                },
              ]
            }
            virtual={undefined}
          >
             
            Criar novo perfil
             
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#FFF",
          "justifyContent": "center",
          "marginTop": 10,
          "padding": 15,
        }
      }
    >
      <View
        style={Object {}}
      >
        <View
          style={
            Object {
              "marginBottom": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#000000",
                "fontSize": 17,
                "fontWeight": "500",
                "marginBottom": 7,
              }
            }
          >
            Matricula
          </Text>
          <View
            style={Object {}}
          >
            <TextInput
              accessibilityLabel="Matricula"
              allowFontScaling={true}
              autoCapitalize={undefined}
              autoCorrect={undefined}
              autoFocus={undefined}
              blurOnSubmit={undefined}
              clearButtonMode={undefined}
              clearTextOnFocus={undefined}
              editable={undefined}
              enablesReturnKeyAutomatically={undefined}
              keyboardAppearance={undefined}
              keyboardType={undefined}
              maxLength={undefined}
              multiline={undefined}
              numberOfLines={undefined}
              onBlur={undefined}
              onChange={undefined}
              onChangeText={[Function]}
              onContentSizeChange={undefined}
              onEndEditing={undefined}
              onFocus={undefined}
              onKeyPress={undefined}
              onLayout={undefined}
              onSelectionChange={undefined}
              onSubmitEditing={undefined}
              placeholder={undefined}
              placeholderTextColor={undefined}
              returnKeyType={undefined}
              secureTextEntry={undefined}
              selectTextOnFocus={undefined}
              selectionColor={undefined}
              selectionState={undefined}
              style={
                Object {
                  "borderColor": "#cccccc",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "color": "#000000",
                  "fontSize": 17,
                  "height": 36,
                  "marginBottom": 5,
                  "paddingHorizontal": 7,
                  "paddingVertical": 7,
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
        <View
          style={
            Object {
              "marginBottom": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#000000",
                "fontSize": 17,
                "fontWeight": "500",
                "marginBottom": 7,
              }
            }
          >
            Especialidade
          </Text>
          <View
            style={Object {}}
          >
            <TextInput
              accessibilityLabel="Especialidade"
              allowFontScaling={true}
              autoCapitalize={undefined}
              autoCorrect={undefined}
              autoFocus={undefined}
              blurOnSubmit={undefined}
              clearButtonMode={undefined}
              clearTextOnFocus={undefined}
              editable={undefined}
              enablesReturnKeyAutomatically={undefined}
              keyboardAppearance={undefined}
              keyboardType={undefined}
              maxLength={undefined}
              multiline={undefined}
              numberOfLines={undefined}
              onBlur={undefined}
              onChange={undefined}
              onChangeText={[Function]}
              onContentSizeChange={undefined}
              onEndEditing={undefined}
              onFocus={undefined}
              onKeyPress={undefined}
              onLayout={undefined}
              onSelectionChange={undefined}
              onSubmitEditing={undefined}
              placeholder={undefined}
              placeholderTextColor={undefined}
              returnKeyType={undefined}
              secureTextEntry={undefined}
              selectTextOnFocus={undefined}
              selectionColor={undefined}
              selectionState={undefined}
              style={
                Object {
                  "borderColor": "#cccccc",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "color": "#000000",
                  "fontSize": 17,
                  "height": 36,
                  "marginBottom": 5,
                  "paddingHorizontal": 7,
                  "paddingVertical": 7,
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "height": "45%",
        }
      }
    >
      <View
        style={
          Object {
            "height": 50,
            "marginBottom": 5,
            "marginLeft": 10,
            "width": 300,
          }
        }
      >
        <RCTPicker
          items={
            Array [
              Object {
                "label": "Tipo de Perfil",
                "textColor": undefined,
                "value": "",
              },
              Object {
                "label": "Funcion√°rio",
                "textColor": undefined,
                "value": "employee",
              },
              Object {
                "label": "Gestor de Setor",
                "textColor": undefined,
                "value": "sector_manager",
              },
              Object {
                "label": "Gestor de Institui√ß√£o",
                "textColor": undefined,
                "value": "institution_manager",
              },
            ]
          }
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          selectedIndex={0}
          style={
            Array [
              Object {
                "height": 216,
              },
              undefined,
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "height": 1334,
          }
        }
        virtual={undefined}
      >
        <RCTScrollView
          automaticallyAdjustContentInsets={false}
          contentContainerStyle={
            Array [
              Object {
                "padding": undefined,
              },
              undefined,
            ]
          }
          contentInset={
            Object {
              "bottom": 0,
            }
          }
          enableAutomaticScroll={true}
          enableResetScrollToCoords={true}
          extraHeight={75}
          extraScrollHeight={0}
          getScrollResponder={[Function]}
          handleOnScroll={[Function]}
          keyboardDismissMode="interactive"
          keyboardOpeningTime={250}
          keyboardShouldPersistTaps="handled"
          keyboardSpace={0}
          onScroll={[Function]}
          resetKeyboardSpace={[Function]}
          resetScrollToCoords={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          scrollEventThrottle={1}
          scrollForExtraHeightOnAndroid={[Function]}
          scrollToEnd={[Function]}
          scrollToFocusedInput={[Function]}
          scrollToPosition={[Function]}
          showsVerticalScrollIndicator={true}
          style={
            Object {
              "backgroundColor": "transparent",
              "flex": 1,
            }
          }
          viewIsInsideTabBar={false}
          virtual={undefined}
        >
          <View>
            <ActivityIndicator
              animating={true}
              color="#5f4b8b"
              hidesWhenStopped={true}
              size="large"
              style={
                Object {
                  "height": 80,
                }
              }
              virtual={undefined}
            />
          </View>
        </RCTScrollView>
      </View>
      <View />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "marginTop": 50,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "height": 1334,
          }
        }
        virtual={undefined}
      >
        <RCTScrollView
          automaticallyAdjustContentInsets={false}
          contentContainerStyle={
            Array [
              Object {
                "padding": undefined,
              },
              undefined,
            ]
          }
          contentInset={
            Object {
              "bottom": 0,
            }
          }
          enableAutomaticScroll={true}
          enableResetScrollToCoords={true}
          extraHeight={75}
          extraScrollHeight={0}
          getScrollResponder={[Function]}
          handleOnScroll={[Function]}
          keyboardDismissMode="interactive"
          keyboardOpeningTime={250}
          keyboardShouldPersistTaps="handled"
          keyboardSpace={0}
          onScroll={[Function]}
          resetKeyboardSpace={[Function]}
          resetScrollToCoords={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          scrollEventThrottle={1}
          scrollForExtraHeightOnAndroid={[Function]}
          scrollToEnd={[Function]}
          scrollToFocusedInput={[Function]}
          scrollToPosition={[Function]}
          showsVerticalScrollIndicator={true}
          style={
            Object {
              "backgroundColor": "transparent",
              "flex": 1,
            }
          }
          viewIsInsideTabBar={false}
          virtual={undefined}
        >
          <View>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "backgroundColor": "#5f4b8b",
                  "borderBottomWidth": null,
                  "borderColor": "#007aff",
                  "borderLeftWidth": null,
                  "borderRadius": 57,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "elevation": 2,
                  "flexDirection": "row",
                  "height": 50,
                  "justifyContent": "center",
                  "marginBottom": 0,
                  "marginTop": 30,
                  "opacity": 1,
                  "paddingBottom": 6,
                  "paddingHorizontal": 50,
                  "paddingTop": 6,
                  "width": 180,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignItems": "center",
                    "color": "white",
                    "fontFamily": "Raleway",
                    "fontSize": 16,
                  }
                }
              >
                Cadastrar
              </Text>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </View>
</View>
`;
